### PR TITLE
Raise error if different than zmq.EAGAIN

### PR DESCRIPTION
**Pull request by ouziel-slama**
* [ ] Double-check the spelling and grammar of all strings, code comments, etc.
* [ ] Double-check that all code is deterministic that needs to be
* [ ] Add tests to cover any new or revised logic
* [ ] Ensure that the test suite passes
* [ ] Update the project [release notes](release-notes/)
* [ ] Update the project documentation, as appropriate, with a corresponding Pull Request in the [Documentation repository](https://github.com/CounterpartyXCP/Documentation)

### CODE REVIEW

File: counterparty-core/counterpartycore/lib/follow.py [hunk 1]
```diff
@@ -182,9 +182,11 @@ async def receive_multipart(self, socket, topic_name):
...
             flags = 0 if topic_name == "sequence" else zmq.NOBLOCK
             topic, body, seq = await socket.recv_multipart(flags=flags)
-        except zmq.ZMQError:
-            logger.trace("No message available in topic `%s`", topic_name)
-            return
+        except zmq.ZMQError as e:
+            if e.errno == zmq.EAGAIN:
+                logger.trace("No message available in topic `%s`", topic_name)
+                return
+            raise e
```
Comment (adamkrellenstein) [inline counterparty-core/counterpartycore/lib/follow.py:189]: if the error gets raised, will the client restart?

**Review by adamkrellenstein — APPROVED** [review]

**Comment by adamkrellenstein** [comment]
(checklist! 🙀)

File: counterparty-core/counterpartycore/lib/follow.py (hunk 1 above)
Comment (ouziel-slama) [reply counterparty-core/counterpartycore/lib/follow.py:190]: If there is an error other than zmq.EAGAIN, we must first identify it to react accordingly.

**Comment by ouziel-slama** [comment]
With this PR we ensure that an error other than zmq.EAGAIN is not silent. This will eventually help identify the problem with this issue: https://github.com/CounterpartyXCP/counterparty-core/issues/1933

**Pull request merged**